### PR TITLE
stgit: enable tests and install documentation

### PR DESCRIPTION
**Pull request by jshholland**
###### Motivation for this change

stgit uses a non-standard build process that `buildPythonApplication` doesn't understand. This PR will customise the build phases to install the man pages for stg and subcommands and run the tests.

I'm not sure whether the build is non-standard enough to justify not using `buildPythonApplication` at all and just using `stdenv.makeDerivationNoCC` or something like that. In particular, I don't like that I have to manually specify `$makeFlags` in each build phase; any tips on how to avoid needing that would be welcome from reviewers.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122370` at 8264b9db run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>stgit</li>
  </ul>
</details>
<details>
  <summary>10 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl' doesn't match anything in file 'Documentation/docbook.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl' doesn't match anything in file 'Documentation/manpage-bold-literal.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl' doesn't match anything in file 'Documentation/manpage.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl' doesn't match anything in file 'Documentation/manpage-quote-apos.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl' doesn't match anything in file 'Documentation/manpage-normal.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl' doesn't match anything in file 'Documentation/manpage-normal.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl' doesn't match anything in file 'Documentation/manpage-quote-apos.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl' doesn't match anything in file 'Documentation/texi.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl-ns/current/manpages/docbook.xsl' doesn't match anything in file 'Documentation/manpage-bold-literal.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl' doesn't match anything in file 'Documentation/texi.xsl'
Near pkgs/applications/version-management/git-and-tools/stgit/default.nix:54:0:
```
   |
54 |     description = "A patch manager implemented on top of Git";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/applications/version-management/git-and-tools/stgit/default.nix [hunk 1]
```diff
@@ -11,16 +20,41 @@ python3Packages.buildPythonApplication rec {
...
+
+  checkTarget = "test";
+  checkFlags = [ "PERL_PATH=${perl}/bin/perl" ];
+
+  installTargets = [ "install" "install-doc" ];
+  postInstall = ''
+    installShellCompletion --cmd stg \
+      $out/share/stgit/completion/stg.fish \
+      $out/share/stgit/completion/stgit.bash \
+      $out/share/stgit/completion/stgit.zsh
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/stgit/default.nix:55]:
```suggestion
      --shell fish $out/share/stgit/completion/stg.fish \
      --shell bash $out/share/stgit/completion/stgit.bash \
      --shel zsh $out/share/stgit/completion/stgit.zsh
```

Comment (jshholland) [reply pkgs/applications/version-management/git-and-tools/stgit/default.nix]: Done, thanks!

File: pkgs/applications/version-management/git-and-tools/stgit/default.nix [hunk 2]
```diff
@@ -11,16 +20,41 @@ python3Packages.buildPythonApplication rec {
...
-  checkInputs = [ git ];
+  format = "other";
 
-  postInstall = ''
-    installShellCompletion $out/share/stgit/completion/stg.fish
-    installShellCompletion --name stg $out/share/stgit/completion/stgit.bash
-    installShellCompletion --name _stg $out/share/stgit/completion/stgit.zsh
+  checkInputs = [ git perl ];
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/stgit/default.nix:29]:
```suggestion
  postPatch = ''
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 122370` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>stgit: <a href="https://termbin.com/1ew3">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/91c40a6159b8f0c7b0750bae25201fe6/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>tests.trivial</li>
</ul>
</details>

**Pull request merged**